src/GC/GC.jl

```diff
@@ -3,77 +3,135 @@
 
 Garbage collection of Python objects.
 
-See `disable` and `enable`.
+See [`enable`](@ref), [`disable`](@ref) and [`gc`](@ref).
 """
 module GC
 
 using ..C: C
 
-const ENABLED = Ref(true)
 const QUEUE = C.PyPtr[]
+const QUEUE_LOCK = Threads.SpinLock()
+const HOOK = WeakRef()
 
 """
     PythonCall.GC.disable()
 
-Disable the PythonCall garbage collector.
+Do nothing.
 
-This means that whenever a Python object owned by Julia is finalized, it is not immediately
-freed but is instead added to a queue of objects to free later when `enable()` is called.
+!!! note
 
-Like most PythonCall functions, you must only call this from the main thread.
+    Historically this would disable the PythonCall garbage collector. This was required
+    for safety in multi-threaded code but is no longer needed, so this is now a no-op.
 """
-function disable()
-    ENABLED[] = false
-    return
-end
+disable() = nothing
```

Comment:
```suggestion
disable() = (Base.depwarn("disabling the PythonCall GC is no longer needed", :disable); nothing)
```

Reply:
`Base.depwarn` is not documented and there doesn't seem to be an equivalent in the API

Reply:
Unresolving this